### PR TITLE
Prototype editing and deleting function

### PR DESCRIPTION
**Pull request by ryo-skw**
# WHAT
Add Prototype editing and deleting function

# WHY
In order to allow users to freely handle their posts

### CODE REVIEW

**Comment by ryo-skw** [comment]
https://gyazo.com/ca80d5b0b86aba6000dc77c567bfba7a
↑Detail screen
https://gyazo.com/91beb2d7e1fab8c4624a3a843ab180b2
↑Edit screen

**Comment by nekoze1210** [comment]
https://gyazo.com/211322cd3c517d9564f9f2adb5518938

File: app/models/prototype.rb [hunk 1]
```diff
@@ -1,7 +1,7 @@
 class Prototype < ActiveRecord::Base
-  has_many :images
+  has_many :images, dependent: :delete_all
```
Comment (nekoze1210) [inline app/models/prototype.rb]: いいね👍

File: app/views/prototypes/edit.html.haml [hunk 2]
```diff
@@ -0,0 +1,39 @@
...
+          = f.text_field :title, placeholder: "Title", class: "proto-new-title"
+      .row
+        .col-md-12
+          .cover-image-upload
+            =f.fields_for :images do |image_form|
+              = image_form.file_field :image
+              = image_form.hidden_field :role, value: :main
+        .col-md-12
+          %ul.proto-sub-list.list-group
+            - 2.times do
```
Comment (nekoze1210) [inline app/views/prototypes/edit.html.haml]: 2.times使う必要はないね
fields_forについてもっと調べてみて！
http://j-caw.co.jp/blog/?p=1590

**Comment by ryo-skw** [comment]
https://gyazo.com/2f4b2eeb8529eb5adef3db2db92bdf5d
↑Detail screen
https://gyazo.com/f48bc4182eb34fb5ae413987573248b9
↑Edit screen

File: app/views/prototypes/edit.html.haml [hunk 3]
```diff
@@ -12,12 +12,11 @@
               = image_form.hidden_field :role, value: :main
         .col-md-12
           %ul.proto-sub-list.list-group
-            - 2.times do
+            = f.fields_for :images, @sub_images do |i|
```
Comment (nekoze1210) [inline app/views/prototypes/edit.html.haml]: NICE

**Comment by nekoze1210** [comment]
LGTM! ドンドン行こう！

**Pull request merged**